test/e2e/destroycluster/destroy_test.go

```diff
@@ -0,0 +1,225 @@
+package destroycluster
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/cache"
+	clientwatch "k8s.io/client-go/tools/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	"github.com/openshift/hive/pkg/install"
+	"github.com/openshift/hive/test/e2e/common"
+)
+
+const (
+	timeout = 30 * time.Minute
+)
+
+func TestDestroyCluster(t *testing.T) {
+	logger := log.WithField("test", "TestDestroyCluster")
+	logger.Info("Starting destroy cluster test")
+
+	cd := common.MustGetClusterDeployment()
+	logger = logger.WithField("cluster", fmt.Sprintf("%s/%s", cd.Namespace, cd.Name))
+	if !cd.Spec.Installed {
+		logger.Warn("Cluster deployment did not finish installing")
+	}
+
+	fail := failTestFunc(t, logger)
+
+	c := common.MustGetHiveClient()
+	logger.Info("Deleting cluster deployment")
+	err := c.HiveV1().ClusterDeployments(cd.Namespace).Delete(context.TODO(), cd.Name, metav1.DeleteOptions{})
+	if err != nil {
+		fail("Failed to delete cluster deployment: %v", err)
+	}
+
+	go monitorDeprovisionJob(cd, logger)
```

Comment:
Can we add a parameter that will get set to true when the uninstall job is seen? The destroy test can then fail if the variable is false at the end, indicating that the uninstall job did not actually run.

Reply:
it would still fail if the clusterdeployment never goes away, no?

Reply:
Yes, but the check is to ensure that Hive is actually running the uninstall job and not just removing the finalizer and leaking resources.

Reply:
Ack, will fix